arch/xtensa/core/gen_zsr.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 Intel corporation
+# SPDX-License-Identifier: Apache-2.0
+import sys
+import re
+
+# Scratch register allocator.  Zephyr uses multiple Xtensa SRs as
+# scratch space for various special purposes.  Unfortunately the
+# configurable architecture means that not all registers will be the
+# same on every device.  This script parses a pre-cooked ("gcc -E
+# -dM") core-isa.h file for the current architecture and assigns
+# registers to usages.
+
+NEEDED = ("ALLOCA", "CPU", "FLUSH")
+
+coreisa = sys.argv[1]
+outfile = sys.argv[2]
+
+syms = {}
```

Comment:
Can you do the usual `if __name__ == "__main__":` dance for import+interactive debug?

Reply:
Heh, uh... no? :)  Just let this be my one bit of iconoclasm.  I genuinely don't understand why python people think it's a good thing to indent an entire file by an extra tab stop completely needlessly and then put a noop function call at the end of the file.  I mean, functional decomposition in a large script with lots of stuff going on?  Sure.  But this is a trivial straight-through logic kind of thing.  We never did that in shell or perl, I don't see why the culture needs to change here.  As this evolves, we can totally move it in that direction if needed.  For for the logic as it stands, I just like it a lot better like this.

Reply:
> We never did that in shell or perl,

Here are a few technical, non-subjective reasons to do this in shell: https://github.com/thesofproject/sof-test/issues/740

> I genuinely don't understand why python people think it's a good thing to indent an entire file...

Most of the reasons above seem to apply to Python the same. I bet you can find more on the Internet.

I agree it's not important for "small" scripts. The size threshold is of course much more subjective; I feel like this one is very slightly crossing the line. No big deal.

Reply:
> I genuinely don't understand why python people think it's a good thing to indent an entire file...

You just made me wonder why it is customary to have a first level of indentation in all functions in all programming languages. In most languages there is little or no code outside functions so the first indentation level adds no information after all.

Maybe it just subconsciously follows standard typography usage https://en.wikipedia.org/wiki/Margin_(typography)#The_Digital_Page

Off topic sorry.